### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by LorDEvilDead**
**new's:**

1. setuped gems: devise, jwt, devise-jwt, b-crypt
2. added registrations_controller
3. added sessions_controller
4. setuped user's model to devise and for current_user
5. added new tests in request spec and model spec

**fixes:**

1. connect user's servises to registration controlle
2. deleted create servise 
3. added new fields to user_factory

### CODE REVIEW

File: backend/app/controllers/api/v1/users/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module Users
+      class SessionsController < Devise::SessionsController
+        include RackSessionsFix
+        # binding.pry
```
Comment (MikhailZaitsau) [inline backend/app/controllers/api/v1/users/sessions_controller.rb]: Unnecessary comment

File: backend/Gemfile [hunk 2]
```diff
@@ -59,3 +59,18 @@ group :development do
...
+gem 'dry-configurable', '1.2.0'
+gem 'dry-core', '1.0.1'
+gem 'jwt', '2.8.2'
+gem 'orm_adapter', '0.5.0'
+gem 'responders', '3.1.1'
+gem 'warden', '1.2.9'
+gem 'warden-jwt_auth', '0.10.0'
+
+gem 'nokogiri', '1.16.6', require: false
+gem 'travel'
```
Comment (MikhailZaitsau) [inline backend/Gemfile]: Is this gem in use?

File: backend/db/migrate/20240807220910_delete_index_from_user.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
```
Comment (MikhailZaitsau) [inline backend/db/migrate/20240807220910_delete_index_from_user.rb]: Remember me, please, why you have decide to delete migration, that have add this index add create new migration, removed this index?

File: backend/spec/factories/users.rb [hunk 4]
```diff
@@ -1,11 +1,12 @@
 # frozen_string_literal: true
 
+require 'pry'
 FactoryBot.define do
   factory :user do
     first_name { Faker::Name.first_name }
     last_name { Faker::Name.last_name }
     email { Faker::Internet.email }
-    encrypted_password { Faker::Internet.password }
+    password { Faker::Internet.password }
```
Comment (MikhailZaitsau) [inline backend/spec/factories/users.rb:9]: Sure, it will work without `password_confirmation`?

File: backend/spec/services/users/update_spec.rb [hunk 5]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 require 'rails_helper'
-
+require 'pry'
```
Comment (MikhailZaitsau) [inline backend/spec/services/users/update_spec.rb]: Looks like unnecessary requirement

File: backend/db/migrate/20240807220910_delete_index_from_user.rb (hunk 3 above)
Comment (LorDEvilDead) [reply backend/db/migrate/20240807220910_delete_index_from_user.rb]: becouse user indexed throught jti
```
  def change
    add_column :users, :jti, :string, null: false, default: ' '
    add_index :users, :jti, unique: true
  end
```
@MikhailZaitsau

File: backend/spec/factories/users.rb (hunk 4 above)
Comment (LorDEvilDead) [reply backend/spec/factories/users.rb:9]: > Sure, it will work without `password_confirmation`?

In next task

**Review by MikhailZaitsau — APPROVED** [review]

**Pull request merged**